### PR TITLE
feat(storage): add sample for creating a dual-region bucket

### LINKED ISSUE #2497: storage: Add Sample for Dual-Region Bucket Creation
**FR:** Add sample for dual-region bucket creation.

The sample for creating a bucket should accept a bucket name and 2 regions (e.g. `my-bucket`, `US-EAST1`, `US-WEST1`).

**Region tag:** `storage_create_bucket_dual_region`

**System tests:**
- Verify a dual region bucket with specified regions can be created
- Verify a dual region bucket's `location` metadata is the specified dual region provided
- Verify a dual region bucket's `locationType` metadata is `dual-region`

[Internal]
- For canonical samples: [go/dual-region-canonical-sample](http://go/dual-region-canonical-sample)
- For details and coordination: [go/gcs-dpe-dual-region-milestones](http://go/gcs-dpe-dual-region-milestones)

### PR DESCRIPTION
**Pull request by BrennaEpp**
Fixes #2497

### CODE REVIEW

File: storage/buckets/create_bucket_dual_region.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+// provided locations.
+func createBucketDualRegion(w io.Writer, projectID, bucketName, region1, region2 string) error {
+	// projectID := "my-project-id"
+	// bucketName := "bucket-name"
+	// region1 := "US-EAST1"
+	// region2 := "US-WEST1"
+
+	ctx := context.Background()
+
+	// Initialize the Storage client that will be used to send requests. This
```
Comment (tritone) [inline storage/buckets/create_bucket_dual_region.go:37]: I don't think this comment is necessary-- we don't have it in other Go samples AFAIK.

File: storage/buckets/buckets_test.go [hunk 2]
```diff
@@ -86,6 +86,32 @@ func TestCreateBucketClassLocation(t *testing.T) {
...
+	if err := createBucketDualRegion(ioutil.Discard, tc.ProjectID, bucketName, region1, region2); err != nil {
+		t.Fatalf("createBucketClassLocation: %v", err)
+	}
+
+	attrs, err := client.Bucket(bucketName).Attrs(ctx)
+	if err != nil {
+		t.Fatalf("Bucket(%q).Attrs: %v", bucketName, err)
+	}
+
+	if !strings.Contains(attrs.Location, region1) || !strings.Contains(attrs.Location, region2) {
```
Comment (tritone) [inline storage/buckets/buckets_test.go:107]: Nit: we should probably verify the exact string here.

**Review by tritone — COMMENTED** [review]
couple small things, overall looks good

**Review by enocom — APPROVED** [review]
LGTM.

The comment above the client is coming from the https://googlecloudplatform.github.io/samples-style-guide/#initializing-clients.

**Review by danielbankhead — APPROVED** [review]

File: storage/buckets/buckets_test.go (hunk 2 above)
Comment (BrennaEpp) [reply storage/buckets/buckets_test.go:107]: The string returned may be either "region1+region2" or "region2+region1"
I received confirmation that the canonical order is deterministic and alphabetical so we could change that here but I'm hesitant to verify against an exact string when the order may be reversed and it means the same thing.

File: storage/buckets/create_bucket_dual_region.go (hunk 1 above)
Comment (BrennaEpp) [reply storage/buckets/create_bucket_dual_region.go:37]: The comment above the client is coming from the https://googlecloudplatform.github.io/samples-style-guide/#initializing-clients.

We don't have this comment in other samples - but we should probably either update the guidance (or overrule it in the Go specific guidance) or follow it going forward?

Comment (BrennaEpp) [reply storage/buckets/create_bucket_dual_region.go]: Sounds good.

**Pull request merged**